### PR TITLE
Prevent mapael from raising an error if the legend container doesn't exist

### PR DESCRIPTION
**Pull request by neveldo**
1 - The option "display" should be checked before checking the existence of a container with a class 'cssClass' (otherwise, users are forced to add a legend container even if just want to display a choropleth without displaying the related legend)

2 - I think that, even if the option 'display' is set to true, we shouldn't rise an error if the container doesn't exist. In fact, some user define their own default options along with some default ones for the legend (in order to customize fonts, colors, etc), and they extend their default options object in order to display both legended and non-legended maps. As the 'display' option is set to true by default within mapael, it will raise an error for  the non-legended maps in this case.

### CODE REVIEW

**Comment by neveldo** [comment]
Hmm, I will check ASAP why my pull-request broke the unit tests pass (don't know if it's related with my update)

File: js/jquery.mapael.js [hunk 1]
```diff
@@ -1709,11 +1709,9 @@
             }
 
             for (var j = 0; j < legendsOptions.length; ++j) {
-                // Check for class existence
-                if (legendsOptions[j].cssClass === "" || $("." + legendsOptions[j].cssClass, self.$container).length === 0) {
-                    throw new Error("The legend class `" + legendsOptions[j].cssClass + "` doesn't exists.");
-                }
-                if (legendsOptions[j].display === true && $.isArray(legendsOptions[j].slices) && legendsOptions[j].slices.length > 0) {
+                if (legendsOptions[j].display === true  && $.isArray(legendsOptions[j].slices) && legendsOptions[j].slices.length > 0
+                    && legendsOptions[j].cssClass !== "" && $("." + legendsOptions[j].cssClass, self.$container).length !== 0
```
Comment (Indigo744) [inline js/jquery.mapael.js]: Issue is here: the `&&` should not after a line break but before.

**Comment by neveldo** [comment]
Hi @Indigo744 ,

You are right, I just saw it in the travis log. It's now fixed (I also fixed the conflict)

**Comment by Indigo744** [comment]
It is a good fix :) I'll go ahead and squash it to master.

**Pull request merged**